src/transformers/modeling_utils.py

```diff
@@ -3392,70 +3392,70 @@ def from_pretrained(
                         )
                         if resolved_archive_file is not None:
                             is_sharded = True
-
-                    if not local_files_only and resolved_archive_file is not None:
-                        if filename in [WEIGHTS_NAME, WEIGHTS_INDEX_NAME]:
-                            # If the PyTorch file was found, check if there is a safetensors file on the repository
-                            # If there is no safetensors file on the repositories, start an auto conversion
-                            safe_weights_name = SAFE_WEIGHTS_INDEX_NAME if is_sharded else SAFE_WEIGHTS_NAME
+                    if not local_files_only:
```

Comment:
Let's check both `local_files_only` (variable passed by the user) and `is_offline_mode()` (from env variable) to prevent any mess. Other than that, looks good to me!

```suggestion
                    if not local_files_only and not is_offline_mode():
```